server/java/src/main/java/com/godaddy/asherah/grpc/AppEncryptionServer.java

```diff
@@ -0,0 +1,207 @@
+  private static int sessionCacheMaxSize;
+  @CommandLine.Option(names = "--session-cache-expire-minutes", defaultValue = "2",
+      description = "Evict the session from cache after some minutes.")
+  private static int sessionCacheExpireMinutes;
+
+  private static SessionFactory sessionFactory;
+
+  private AppEncryptionServer() {
+  }
+
+  private NettyServerBuilder bindUDS(final String uds) {
+
+    // Create a new server to listen on a socket
+    EventLoopGroup group;
+    NettyServerBuilder builder = NettyServerBuilder.forAddress(new DomainSocketAddress(uds));
+    if (SystemUtils.IS_OS_MAC) {
+      group = new KQueueEventLoopGroup();
+      builder.channelType(KQueueServerDomainSocketChannel.class);
+    }
+    else if (SystemUtils.IS_OS_LINUX) {
+      group = new EpollEventLoopGroup();
+      builder.channelType(EpollServerDomainSocketChannel.class);
+    }
+    else {
+      throw new IllegalStateException("binding to unix:// addresses is only supported on Linux and macOS");
+    }
+    builder.workerEventLoopGroup(group).bossEventLoopGroup(group);
+
+    return builder;
+  }
+
+  private Metastore<JSONObject> setupMetastore() {
+
+    if (metastoreType == MetastoreType.JDBC) {
+      if (jdbcUrl != null) {
+        logger.info("using JDBC-based metastore...");
+
+        // Setup JDBC persistence from command line argument using Hikari connection pooling
+        HikariDataSource dataSource = new HikariDataSource();
+        dataSource.setJdbcUrl(jdbcUrl);
+        return JdbcMetastoreImpl.newBuilder(dataSource).build();
+      }
+      else {
+        CommandLine.usage(this, System.out);
+        return null;
+      }
+    }
+    else if (metastoreType == MetastoreType.DYNAMODB) {
+      logger.info("using DynamoDB-based metastore...");
+
+      return DynamoDbMetastoreImpl.newBuilder().build();
+    }
+
+    logger.info("using in-memory metastore...");
+    return new InMemoryMetastoreImpl<>();
+  }
+
+  private CryptoPolicy setupCryptoPolicy() {
+
+    return BasicExpiringCryptoPolicy
+        .newBuilder()
+        .withKeyExpirationDays(keyExpirationDays)
+        .withRevokeCheckMinutes(revokeCheckMinutes)
+        .withCanCacheSessions(sessionCacheEnabled)
+        .withSessionCacheMaxSize(sessionCacheMaxSize)
+        .withSessionCacheExpireMinutes(sessionCacheExpireMinutes)
+        .build();
+  }
+
+  private KeyManagementService setupKeyManagementService() {
+
+    if (kmsType == KmsType.AWS) {
+      if (preferredRegion != null && regionMap != null) {
+        logger.info("using AWS KMS...");
+
+        // build the ARN regions including preferred region
+        return AwsKeyManagementServiceImpl
+          .newBuilder(regionMap, preferredRegion).build();
+      }
+      else {
+        CommandLine.usage(this, System.out);
+        return null;
+      }
+    }
+
+    logger.info("using static KMS...");
+    return new StaticKeyManagementServiceImpl("mysupersecretstaticmasterkey!!!!");
```

Comment:
Can flatten and simplify by inverting the logic

```suggestion
    if (kmsType == KmsType.STATIC) {
      logger.info("using static KMS...");
      return new StaticKeyManagementServiceImpl("mysupersecretstaticmasterkey!!!!");
    }

    if (preferredRegion == null || regionMap == null) {
      CommandLine.usage(this, System.out);
      return null;
    }

    logger.info("using AWS KMS...");
   
    // build the ARN regions including preferred region
    return AwsKeyManagementServiceImpl
      .newBuilder(regionMap, preferredRegion).build();
```